Backend/Backend/Services/SocialService.cs

```diff
@@ -482,4 +501,91 @@ public async Task<bool> RemoveReviewFromPodcastAsync(Guid podcastId, User user)
     }
 
     #endregion
+
+
+    #region Points
+
+    /// <summary>
+    /// Endpoints to gift Points to a particular episode
+    /// </summary>
+    /// <param name="points"></param>
+    /// <param name="episodeId"></param>
+    /// <param name="user"></param>
+    /// <returns></returns>
+    /// <exception cref="Exception"></exception>
+
+    public async Task<string> GiftPoints(int points, Guid episodeId, User user)
+    {
+        // Check whether the point is greater then zero or Not
+        if(points <= 0)
+        {
+            throw new Exception("No of Points should be Greater then zero");
+
+        }
+        // Limit the points to 200
+        if(points > 200)
+        {
+            throw new Exception("You cant gift more then 200 points");
+        }
+
+        // Check whether the person gifting is different person not the podcaster
+        Episode? ep1 = await _db.Episodes.Include(u => u.Podcast).FirstOrDefaultAsync(u => u.Podcast.PodcasterId == user.Id);
+
+        // if he is a podcaster then throw an error
+        if(ep1 != null)
+        {
+            throw new Exception("You cant gift points to yourself");
+
+        }
+
+        // If all is good then add points to the DB
+        Guid PointId = Guid.NewGuid();
+        
+        await _db.Points.AddAsync(new Points { Id = PointId, EpisodeId = episodeId, UserId = user.Id, PointCount = points,CreatedAt = DateTime.Now, Amount = points * POINT_CONVERSION });
```

Comment:
wait does this mean that when you gift points, they only get a tenth of it?!?!

Reply:
So each point is 0.1 CAD so if you gift 10 points then its 1CAD